### PR TITLE
feat: `public_key` method added to `KeyfileAccount` class [APE-1466]

### PR DESCRIPTION
**Pull request by Aviksaikat**
### What I did

fixes: #1701

### How to verify it

```py
from ape import accounts
a = accounts[0]
a.public_key
# Enter your password
>> HexBytes("0x33abababbababababbabababababababba..")
```

### Checklist

- [x] All changes are completed
- [x] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape_accounts/accounts.py [hunk 1]
```diff
@@ -92,6 +93,17 @@ def __key(self) -> HexBytes:
 
         return key
 
+    @property
+    def public_key(self) -> HexBytes:
+        key = self.__key
+
+        # Derive the public key from the private key
+        pk = keys.PrivateKey(key)
```
Comment (antazoey) [inline src/ape_accounts/accounts.py:101]: could we save this to the account's JSON so we don't have to decrypt it like this? it is odd to require a private key to get a public key. for accounts that dont yet have it in the JSON, we could do this approach here and the save it in the JSON so next time decrypting is no required.

**Review by antazoey — COMMENTED** [review]
looks good, but the public key should maybe just be in the JSON unencrypted so your passphrase isnt required to get the public key.

thoughts anyone?

File: src/ape_accounts/accounts.py [hunk 2]
```diff
@@ -92,6 +93,17 @@ def __key(self) -> HexBytes:
 
         return key
 
+    @property
+    def public_key(self) -> HexBytes:
```
Comment (antazoey) [inline src/ape_accounts/accounts.py:97]:
```
    @property
    def public_key(self) -> HexBytes:
        if "public_key" in self.keyfile:
            return self.keyfile["public_key"]

        # Derive public key
        passphrase = self._prompt_for_passphrase(default="")
        key = self.__get_key(passphrase=passphrase)
        self.keyfile_path.unlink()

        pk = keys.PrivateKey(key)
        public_key = str(pk.public_key)
        data = json.dumps(EthAccount.encrypt(key, passphrase))
        data["public_key"] = pk

        self.keyfile_path.write_text(json.dumps(data))

        return pk
```

File: src/ape_accounts/accounts.py [hunk 3]
```diff
@@ -92,6 +93,17 @@ def __key(self) -> HexBytes:
...
+        key = self.__key
+
+        # Derive the public key from the private key
+        pk = keys.PrivateKey(key)
+        # convert from eth_keys.datatypes.PublicKey to str to make it HexBytes
+        publicKey = str(pk.public_key)
```
Comment (antazoey) [inline src/ape_accounts/accounts.py:103]: keep_it_snake_case

File: src/ape_accounts/accounts.py (hunk 2 above)
Comment (antazoey) [reply src/ape_accounts/accounts.py:97]: but this requires changing `__key` to `__get_key(passphrase=None)`

**Comment by fubuloubu** [comment]
> looks good, but the public key should maybe just be in the JSON unencrypted so your passphrase isnt required to get the public key.
> 
> thoughts anyone?

can do that, the keyfile format just needs a minimum number of members to work, you can add more

**Review by fubuloubu — APPROVED** [review]

**Comment by Aviksaikat** [comment]
I have opened an issue in the `eth_keys` repo as well so that we can it can be a general property of the `keyfile`. But the repo seems to be very inactive.

**Review by antazoey — APPROVED** [review]
LGTM! This is the best we can do until eth-keyfile updates

**Pull request merged**